stan/math/prim/mat/fun/to_matrix.hpp

```diff
@@ -70,22 +69,46 @@ namespace stan {
 
 
     /**
-     * Returns a matrix representation of the standard vector of standard vectors
-     * of integers with the same dimensions and indexing order.
+     * Returns a matrix representation of a standard vector of Eigen row vectors
+     * with the same dimensions and indexing order.
      *
-     * @param vec vector of vectors of integer values
+     * @tparam T type of the scalar
+     * @param x Eigen vector of vectors of scalar values
      * @return the matrix representation of the input
      */
+    template <typename T>
+    inline Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>
+    to_matrix(const std::vector<Eigen::Matrix<T, 1, Eigen::Dynamic> >& x) {
+      size_t rows = x.size();
+      if (rows != 0) {
```

Comment:
Generally we want the unusual case to come first and return and the usual case to not be nested in an else.  This was a biggy in the Refactoring book and I think it makes code a lot cleaner and easier to follow.